src/YOUR_NAME_ex/ex04/myrand.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Isak Finnøy'
+__email__ = 'isfi@nmbu.no'
+
+
+class LCGRand:
+
+    def __init__(self, seed):
+        """
+        :param seed: arbitrary input
+        """
+        self.seed = seed
+        self.a = 7 ** 5
+        self.m = 2 ** 31 - 1
+
+    def rand(self):
+        """
+        Calculates next random number with the LCG method
+        """
+        self.seed = self.a * self.seed % self.m
+        return self.seed
+
+
+class ListRand:
+
+    def __init__(self, lis):
+        """
+        :param lis: arbitrary list
+        """
+        self.lis = lis
+        self.idx = 0
+        self.len = len(self.lis) - 1
+
+    def rand(self):
+        """
+        Returns next "random" number from an arbitrary list
+        """
+        if self.idx > self.len:

```

Comment:
Instead of naming `len`, which can cause misunderstanding, you can name it `last_idx`, which is the real meaning of this variable.